app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -103,10 +103,27 @@ def reload_answer
   # Publish all the graded submissions.
   def publish_all
     authorize!(:publish_grades, @assessment)
-    graded_submissions = @assessment.submissions.with_graded_state
-    if !graded_submissions.empty?
+    graded_submission_ids = @assessment.submissions.with_graded_state.by_users(course_user_ids).pluck(:id)
+    if graded_submission_ids.empty?
+      head :ok
+    else
```

Comment:
It would be ideal to just change this to `end`, then the subsequent block of code can be indented by 1 less level.

Reply:
Noted